internal/schemautil/custom_diff.go

```diff
@@ -173,3 +175,70 @@ func CustomizeDiffCheckStaticIpDisassociation(_ context.Context, d *schema.Resou
 	// TODO: Check that we block deletions that will result in too few static ips for the plan
 	return nil
 }
+
+// CustomizeDiffDisallowMultipleManyToOneKeys checks that we don't have multiple keys that are going to be converted to
+// a single key in the API request, e.g. 'ip_filter' and 'ip_filter_object' in the same diff.
+func CustomizeDiffDisallowMultipleManyToOneKeys(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	ks := d.GetRawConfig().AsValueMap()
```

Comment:
We actually never use this variable other than as an argument for a loop. Can we do something like?:  
```
for k, v := range d.GetRawConfig().AsValueMap() {
...

// the same can be done here
for k, v := range  v.AsValueSlice()[0].AsValueMap() {
```

Reply:
this increases code complexity

Reply:
I think a bit differently, I always tend to check where the variable is used, and since we only use it for things like: 
```
nks := v.AsValueSlice()[0].AsValueMap()
tks = typedKeys(nks)
```
I vote for something like: 
```
tks = typedKeys(v.AsValueSlice()[0].AsValueMap())
```

The less lines of code the better :)